google/cloud/storage/grpc_plugin.cc

```diff
@@ -15,19 +15,23 @@
 #include "google/cloud/storage/grpc_plugin.h"
 #include "google/cloud/storage/internal/grpc_client.h"
 #include "google/cloud/storage/internal/hybrid_client.h"
+#include "google/cloud/internal/getenv.h"
 
 namespace google {
 namespace cloud {
 namespace storage_experimental {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
 google::cloud::storage::Client DefaultGrpcClient(Options opts) {
-  opts = google::cloud::storage::internal::DefaultOptionsGrpc(std::move(opts));
-  auto config = opts.get<GrpcPluginOption>();
+  using ::google::cloud::internal::GetEnv;
+  auto const config = GetEnv("GOOGLE_CLOUD_CPP_STORAGE_GRPC_CONFIG")
+                          .value_or(opts.get<GrpcPluginOption>());
   if (config == "none" || config.empty()) {
     return google::cloud::storage::Client(std::move(opts));
   }
   if (config == "metadata") {
+    opts =
+        google::cloud::storage::internal::DefaultOptionsGrpc(std::move(opts));
```

Comment:
I am confused. Is the point that we do not want to `DefaultOptionsGrpc` if `config == "media"` ? I would have guessed that `HybridClient` wants the options defaulted.

Also, pulling the env var stuff out of `DefaultOptionsGrpc()` means we have no test that the env var overrides the `GrpcPluginOption`... that seems less good to me.

Reply:
Sorry, I just read the PR description. You explained the point.

It would still be nice if we could unit test the env var behavior. Can we default the authority option differently within `DefaultOptionsGrpc()` depending on the config?

Reply:
> Sorry, I just read the PR description. You explained the point.
> 
> It would still be nice if we could unit test the env var behavior. Can we default the authority option differently within `DefaultOptionsGrpc()` depending on the config?

That would spread the knowledge of what `GrpcPluginOption` does across multiple files.  Does not seem very desirable.  I created a new unit test for the `DefaultGrpcClient()` function.